uniswap-v2-core/test/UniswapV2Pair.spec.ts

```diff
@@ -437,4 +439,62 @@ describe('UniswapV2Pair', () => {
     expect(await token0.balanceOf(pair.address), "Token 0 balance of pair").to.eq(bigNumberify(1000).add(token0ExpBalVexchange))
     expect(await token1.balanceOf(pair.address), "Token 1 balance of pair").to.eq(bigNumberify(1000).add(token1ExpBalVexchange))
   })
+
+
+  /**
+   *  recoverToken - error handling for invalid tokens 
+   */
+  it('recoverToken:invalidToken', async () => {
+    await expect(pair.recoverToken(token0.address)).to.be.revertedWith('Vexchange: INVALID_TOKEN_TO_RECOVER')
+    await expect(pair.recoverToken(token1.address)).to.be.revertedWith('Vexchange: INVALID_TOKEN_TO_RECOVER')
+  
+    // address is a random invalid token address
+    await expect(pair.recoverToken("0x3704E657053C02411aA2Fd0599e75C3d817F81BC")).to.be.reverted
+  })
+
+  /**
+   *  recoverToken - failure when recoverer is AddressZero or not set
+   *  Or when the token address is not a valid ERC20 address
+   */
+  it('recoverToken:AddressZero', async () => {
+    
+    // recoverer should be AddressZero by default
+    expect(await pair.recoverer()).to.eq(AddressZero)
+    await expect(pair.recoverToken(AddressZero)).to.be.revertedWith('Vexchange: RECOVERER_ZERO_ADDRESS')
+
+    let recoveryAddress = other.address
+    await factory.setRecovererForPair(pair.address, recoveryAddress)
+    expect(await pair.recoverer()).to.eq(recoveryAddress);
+  })
+
+  /**
+   *  recoverToken - when there are no tokens to be recovered
+   */
+  it('recoverToken:noAmount', async () => {
+    let recoveryAddress = other.address
+    
+    // There should not be any token of the kind to be recovered
+    // in the recoverer's account
+    expect(await token2.balanceOf(recoveryAddress)).to.be.eq(0)
+    await factory.setRecovererForPair(pair.address, recoveryAddress)
+    expect(await token2.balanceOf(recoveryAddress)).to.eq(0)
```

Comment:
Difference in `to.be.eq` and `to.eq`?

Reply:
I believe they are equivalent and exist in an attempt at approximating more natural language in the code ... so its not a bi deal but we should just standardise style wise - I think most of the code uses the shorter form, so we can just drop the 'be'.

Reply:
yes they're equivalent. I'll change it to the shorter form.